ecl/eclcmd/queries/ecl-queries.cpp

```diff
@@ -368,34 +398,16 @@ class EclCmdQueryFiles : public EclCmdCommon
         if (ret == 0)
         {
             IArrayOf<IConstFileUsedByQuery> &files = resp->getFiles();
-            if (!files.length())
-                fputs("No files used.\n", stdout);
+            IArrayOf<IConstQueryFilesUsed> &queries = resp->getQueries();
+            if (optQuery.length())
+                outputQueryFiles(optQuery.str(), resp->getFiles(), resp->getSuperFiles());
             else
-                fputs("Files used:\n", stdout);
-            ForEachItemIn(i, files)
             {
-                IConstFileUsedByQuery &file = files.item(i);
-                StringBuffer line("  ");
-                line.append(file.getFileName()).append(", ");
-                line.append(file.getFileSize()).append(" bytes, ");
-                line.append(file.getNumberOfParts()).append(" part(s)\n");
-                fputs(line, stdout);
-            }
-            fputs("\n", stdout);
-
-            IArrayOf<IConstQuerySuperFile> &superfiles = resp->getSuperFiles();
-            if (superfiles.length())
-            {
-                fputs("SuperFiles used:\n", stdout);
-                ForEachItemIn(sp, superfiles)
+                ForEachItemIn(i, resp->getQueries())
                 {
-                    IConstQuerySuperFile &superfile = superfiles.item(sp);
-                    fprintf(stdout, "  %s\n", superfile.getName());
-                    StringArray &subfiles = superfile.getSubFiles();
-                    ForEachItemIn(sb, subfiles)
-                        fprintf(stdout, "    > %s\n", subfiles.item(sb));
+                    IConstQueryFilesUsed &query = resp->getQueries().item(i);
+                    outputQueryFiles(query.getQueryId(), query.getFiles(), query.getSuperFiles());
                 }
-                fputs("\n", stdout);
```

Comment:
question: is it possible that the optQuery is not set and the resp->getQueries() returns empty? If yes, should we output a message similar to "No files used.\n" at that situation?

Reply:
Sure, I will add a message.